controllers/awsmachine_tags.go

```diff
@@ -62,6 +70,35 @@ func (r *AWSMachineReconciler) ensureTags(svc service.EC2MachineInterface, machi
 	return changed, nil
 }
 
+// Ensure that the tags of the volumes in the machine are correct
+// Returns bool, error
+// Bool indicates if changes were made or not, allowing the caller to decide
+// if the machine should be updated.
+func (r *AWSMachineReconciler) ensureVolumeTags(svc service.EC2MachineInterface, machine *infrav1.AWSMachine, volume *infrav1.Volume, instanceID *string, additionalTags map[string]string) (bool, error) {
+	annotations := make(map[string]interface{})
```

Comment:
Updating the annotation in this function means the annotation will get overwritten with every EC2 volume.
Either, move the annotation up a level into the calling function, as we end up with
`map[string]map[string]interface{}` or, instead store a separate annotation per volume.

Reply:
does the new changes look ok? I have moved annotation outside to the calling func now